libraries/dotnet/WorkbenchConnector/WorkbenchConnector.cs

```diff
@@ -439,8 +439,12 @@ public virtual void Init()
                 return;
             }
 
-            // Example: http://[::]:64351
-            string first = feat.Addresses.First().Replace("[::]", "host", StringComparison.OrdinalIgnoreCase);
+            // Example: http://[::]:64351 - Prefer non-HTTPS to avoid cert validation errors
+            string first = feat.Addresses.Any(x => !x.Contains("https:", StringComparison.OrdinalIgnoreCase))
+                ? feat.Addresses.First(x => !x.Contains("https:", StringComparison.OrdinalIgnoreCase)).Replace("[::]", "host", StringComparison.OrdinalIgnoreCase)
+                : feat.Addresses.First().Replace("[::]", "host", StringComparison.OrdinalIgnoreCase);
```

Comment:
@tommasodotNET this should fix the issue with port 5001 - e.g. if you're running an agent with 5000 (http) and 5001 (https), the code will select 5000.